test/leak.test.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+var request = require('../').defaults({ json: true });;
+var t = require('chai').assert;
+
+describe('Information Leak', function () {
+
+    it('should not forward cookie headers when the request has a redirect', function (done) {
+
+        request({
+            url: 'https://httpbingo.org/cookies?url=https://google.com/',
+            headers: {
+                'Content-Type': 'application/json',
+                'cookie': 'ajs_anonymous_id=1234567890',
+                'authorization': 'Bearer eyJhb12345abcdef'
+            }
+        }, function (err, response, body) {
+            t.strictEqual(Object.keys(body).length, 0);
```

Comment:
this only tests that the request response is empty, not that a redirect occurred and google.com received the authorization header.

Reply:
see my comment https://github.com/FGRibreau/node-request-retry/pull/138#issuecomment-1043046483

Reply:
@FGRibreau change:

this.options = sanitizeHeaders(options);

To:

this.options = options;

In your index.js and see the test fail.

Your first test API endpoint return the cookie in the response (or empty object if missing), the second one returns bearer token (authorization) or an error code if missing.

If the response body is empty, it means that the function as successfully removed the cookie from the header before making the request so the test succeeded.

You do not require an open redirect for the test, just an URL in the querry parameters. And make the request to an API endpoint which sends you back the headers in the response.

Reply:
here is the test file with open redirect:

```js
'use strict';

var request = require('../').defaults({ json: true });;
var t = require('chai').assert;

describe('Information Leak', function () {

    it('should not forward cookie headers when the request has a redirect', function (done) {

        request({
            url: 'https://httpbingo.org/redirect-to?url=http://httpbingo.org/cookies',
            headers: {
                'Content-Type': 'application/json',
                'cookie': 'ajs_anonymous_id=1234567890',
                'authorization': 'Bearer eyJhb12345abcdef'
            }
        }, function (err, response, body) {
            t.strictEqual(Object.keys(body).length, 0);
            done();
        });
    });

    it('should not forward authorization headers when the request has a redirect', function (done) {

        request({
            url: 'https://httpbingo.org/redirect-to?url=http://httpbingo.org/bearer',
            headers: {
                'Content-Type': 'application/json',
                'cookie': 'ajs_anonymous_id=1234567890',
                'authorization': 'Bearer eyJhb12345abcdef'
            }
        }, function (err, response, body) {
            t.strictEqual(body, undefined);
            done();
        });
    });

});
```

Reply:
Now I can confirm that the test works with that code :)

Reply:
Merci d'avoir pris le temps daccepter mon pull request🙂🥳